experiment/simulation/js/clock.js

```diff
@@ -0,0 +1,170 @@
+import { gates } from "./gate.js";
+import { registerGate } from "./main.js";
+import { setPosition } from "./layout.js";
+
+'use strict';
+
+window.simulate = 0;
+
+export class Clock {
+    constructor(frequency, dutyCycle) {
+        this.frequency = frequency;
+        this.dutyCycle = dutyCycle;
+        this.isOn = false;
+        this.name = "C" + window.numComponents;
+        this.id = "Clock-" + window.numComponents++;
+        this.component = `<div class="low" id=${this.id}><a>0</a><p>${this.name}</p></div>`
+        this.output = false;
+        this.isInput = true;
+        this.inputPoints = [];
+        this.inputs = [];
+        this.outputPoints = [];
+        this.isConnected = false;
+        this.type = "Clock";
+    }
+
+    setId(id) {
+        this.id = id;
+    }
+
+    setName(name) {
+        this.name = name;
+    }
+
+    updateComponent() {
+        this.component = `<div class="low" id=${this.id}><a>0</a><p>${this.name}</p></div>`
+    }
+
+
+    registerComponent(workingArea, x = 0, y = 0) {
+
+         // get width of working area
+         const width = document.getElementById(workingArea).offsetWidth;
+         const height = document.getElementById(workingArea).offsetHeight;
+         let scale = 900;
+         let yScale = 800;
+         x = (x / scale) * width;
+         y = (y / yScale) * height;
+ 
+         const el = document.getElementById(this.id);
+         el.style.left = x + "px";
+         el.style.top = y + "px";
+ 
+         if (this.type !== "Input" && this.type !== "Output" && this.type !== "Clock") {
```

Comment:
I am not sure why we need this line. Please let me know during the next meeting

Reply:
Removed it